File: Week 2 - Arrays/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string sample = get_string("Text: ");
    int letters = count_letters(sample);
    int words = count_words(sample);
    int sentences = count_sentences(sample);

    // For testing purposes, count the letters, words, and sentences in the text.
    // printf("Letters: %i\n", letters);
    // printf("Words: %i\n", words);
    // printf("Sentences: %i\n", sentences);

    // Count the average number of letters per 100 words in the text
    float L = letters / (words / 100.0);

    // Count the average number of sentences per 100 words in the text
    float S = sentences / (words / 100.0);

    // Determine the Coleman-Liau index
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // printf("Grade %i\n", index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int letter_count = 0;
    int length = strlen(text);
    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]))
        {
            letter_count++;
        }
    }
    return letter_count;
}

int count_words(string text)
{
    int word_count = 1;
    int length = strlen(text);
    for (int i = 0; i < length; i++)
    {
        if (text[i] == ' ')
        {
            word_count++;
        }
    }
    return word_count;
}

int count_sentences(string text)
{
    int sentence_count = 0;
    int length = strlen(text);
    for (int i = 0; i < length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence_count++;
        }
    }
    return sentence_count;
}

File: Week 6 - Python/sentimental-readability/readability.py
# Implement a program that computes the approximate grade level needed to comprehend some text,
# per https://cs50.harvard.edu/x/2022/psets/6/readability/ and https://cs50.harvard.edu/x/2022/psets/2/readability/

text = input("Text: ")

# Count the letters in the sample text.
# You may assume that a letter is any lowercase character from a to z or any uppercase character from A to Z.

letters = 0
for letter in text:
    if letter.isalpha():
        letters += 1
# print(f"Letters: {letters}")

# Count the number of words in the sample text.
# For the purpose of this problem, we’ll consider any sequence of characters separated by a space to be a word
# (so a hyphenated word like "sister-in-law" should be considered one word, not three).

words = 1
for letter in text:
    if letter == " ":
        words += 1
# print(f"Words: {words}")

# Count the number of sentences in the sample text.
# You should consider any sequence of characters that ends with a . or a ! or a ? to be a sentence.

sentences = 0
for letter in text:
    if letter in [".", "!", "?"]:
        sentences += 1
# print(f"Sentences: {sentences}")

# Calculate the average number of letters per 100 words in the text
l = (letters / words) * 100

# Calculate the average number of sentences per 100 words in the text
s = (sentences / words) * 100

index = 0.0588 * l - 0.296 * s - 15.8

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")
